pkg/features/kube_features.go

```diff
@@ -801,6 +801,14 @@ const (
 	// Allow users to specify whether to take nodeAffinity/nodeTaint into consideration when
 	// calculating pod topology spread skew.
 	NodeInclusionPolicyInPodTopologySpread featuregate.Feature = "NodeInclusionPolicyInPodTopologySpread"
+
+	// owner: @mimowo
+	// kep: http://kep.k8s.io/3329
+	// alpha: v1.25
+	//
+	// Allow users to specify handling of pod failures based on container exit codes
+	// and pod end state.
+	JobPodFailurePolicy featuregate.Feature = "JobPodFailurePolicy"
```

Comment:
since this feature is primarily controlling whether pod-controllers add conditions, I'd suggest a less job-centric name

Reply:
I'll defer to @alculquicondor on the feature name, but something more pod-centric like that seems better to me

Reply:
Should we consider two feature gates?

`PodDisruptionConditions` and `JobPodFailurePolicy`?

If we go with one, I would go with `PodDisruptionConditions`

Reply:
Yes, I think you need two feature gates.

Reply:
Makes sense, I have added `PodDisruptionConditions` gate for this PR (`JobPodFailurePolicy` remains to control the job_controller PR). Also, the KEP PR is updated accordingly: https://github.com/kubernetes/enhancements/pull/3438

Reply:
The introduction of the second feature gate introduced a merge conflict between the two PRs. Should we consider declaration of the two feature gates on both PRs so that they merge without conflicts once lgtmed/approved?